Improve embedded schemas guide (#3964)
elixir-ecto/ecto: guides/introduction/Embedded Schemas.md

end
```

> Defining embedded schema in such a way is called inline-defined embedded schemas which are:

Reviewer: ```suggestion
Embedded schemas defined in such way are said to be defined inline, which:

```

end
```

Defining embedded schema in such a way is called inline-defined embedded schemas which are:

> - the simpliest way to define embedded schema

Reviewer 2: ```suggestion
```
I would personally take this out, given the other use case that is highlighted further down:

> Like was said before, they are persistence agnostic. This makes embedded schemas ideal for scenarios where you want to manage structured data without necessarily persisting it.

So it's only the simplest if you aren't doing the above.

Author: That's reasonable point of view. Besides this I have a typo in this line, so I'll totally remove it, like you suggested.
Thank you.

end
```

> Defining embedded schema in such a way is called explicit-defined embedded schemas, which:

Reviewer: ```suggestion
Embedded schemas defined in such way are said to be explicit-defined, which:

```

- are dedicated modules having own scope, changeset functions, props, documentation, etc...
- could be embedded by multiple parent schemas
- are persistence agnostic, which means that `embedded_schema` doesn't require to be persisted

> It is important to remember that `embedded_schema` has many use cases independent of `embeds_one` and `embeds_many`. Like was said before, they are persistence agnostic. This makes embedded schemas ideal for scenarios where you want to manage structured data without necessarily persisting it. For example, if you want to build a contact form, you still want to parse and validate the data, but the data is likely not persisted anywhere. Instead, it is used to send an email. Embedded schemas would be a good fit for such a use case.

Reviewer: ```suggestion
It is important to remember that `embedded_schema` has many use cases independent of `embeds_one` and `embeds_many`. As they are persistent agnostic, they are ideal for scenarios where you want to manage structured data without necessarily persisting it. For example, if you want to build a contact form, you still want to parse and validate the data, but the data is likely not persisted anywhere. Instead, it is used to send an email. Embedded schemas would be a good fit for such a use case.
```
